lighthouse-core/test/computed/third-party-summary-test.js

```diff
@@ -0,0 +1,168 @@
+      context.settings.budgets = [{
+        path: '/',
+        options: {
+          // Matches example.com and cdn.example.com
+          firstPartyHostnames: ['*.example.com'],
+        },
+      }];
+      const result = await ComputedThirdPartySummary.request(artifacts, context);
+      expect(result.size).toBe(allResourcesSize - 30 - 10);
+      expect(result.count).toBe(allResourcesCount - 2);
+    });
+
+    it('handles root domain hostname correctly', async () => {
+      context.settings.budgets = [{
+        path: '/',
+        options: {
+          // Matches example.com; does not match cdn.example.com
+          firstPartyHostnames: ['example.com'],
+        },
+      }];
+      const result = await ComputedThirdPartySummary.request(artifacts, context);
+      expect(result.size).toBe(allResourcesSize - 30);
+      expect(result.count).toBe(allResourcesCount - 1);
+    });
+
+    it('handles multiple hostnames correctly', async () => {
+      context.settings.budgets = [{
+        path: '/',
+        options: {
+          firstPartyHostnames: ['example.com', 'my-cdn.com'],
+        },
+      }];
+      const result = await ComputedThirdPartySummary.request(artifacts, context);
+      expect(result.size).toBe(allResourcesSize - 30 - 25);
+      expect(result.count).toBe(allResourcesCount - 2);
+    });
+
+    it('handles syntactical duplication of hostnames', async () => {
+      context.settings.budgets = [{
+        path: '/',
+        options: {
+          firstPartyHostnames: ['my-cdn.com', 'my-cdn.com', 'my-cdn.com'],
+        },
+      }];
+      const result = await ComputedThirdPartySummary.request(artifacts, context);
+      expect(result.size).toBe(allResourcesSize - 25);
+      expect(result.count).toBe(allResourcesCount - 1);
+    });
+
+    it('handles logical duplication of hostnames', async () => {
+      context.settings.budgets = [{
+        path: '/',
+        options: {
+          firstPartyHostnames: ['example.com', '*.example.com', 'cdn.example.com'],
+        },
+      }];
+      const result = await ComputedThirdPartySummary.request(artifacts, context);
+      expect(result.size).toBe(allResourcesSize - 30 - 10);
+      expect(result.count).toBe(allResourcesCount - 2);
+    });
+
+    it('handles using top-level domains as firstPartyHostnames correctly', async () => {
+      context.settings.budgets = [{
+        path: '/',
+        options: {
+          firstPartyHostnames: ['*.com'],
+        },
+      }];
+      const result = await ComputedThirdPartySummary.request(artifacts, context);
+      expect(result.size).toBe(0);
+      expect(result.count).toBe(0);
+    });
+
+    /**
+     * Data URLs should be ignored to avoid double counting.
+     * The filesize of a data URL is reflected in the filesize of the resource that contains it.
+     * A data URL does not result in a separate network request.
+     */
+    it('ignores data URLs', async () => {
```

Comment:
I wonder if we should exclude all of the `NON_NETWORK_PROTOCOLS` in `url-shim`? I think `blob:` is the only other one that would impact things here though we might already be covering that with the `hostname === ''` check

probably worth a test at least to call it out